### DISCUSSION TITLE
Some way to change the front-end assets path?

### DISCUSSION THREAD

**Discussion by FHU-yezi** [flet-dev/flet#389] [General]
Due to the GFW and something else, the speed of load a flet applications in China is a bit slow. Just checked Developer Tools and found we spent most time on loading ttf and js files.

Maybe I can host all the assets on Object Storage Service such as Aliyun OSS or Amazon S3, and use CDN to speed up our apps.

so can I config the path of assets in application code?

**Comment by FeodorFitsner** [comment]
Couldn't be done with some reverse proxy like NGINX? Or you mean hosting Flet web app on a CDN and a Python app server somewhere elase? Could you please elaborate?

**Comment by FHU-yezi** [comment]
All of the front-end assets of flet is hosted with a CDN service, doesn't it?

I mean due to some reason, China goverment has built a network firewall, which called GFW ( Great Fire Wall ) to ban some foreignal websites which maybe harmful to Chinese.

So I can't access the CDN which host the assets and should waiting for quite a long time when the App is loading.

I can host all the assets on a server in China, and they should speed up the loading process. My question is, **how to 'tell' flet to load assets from my server, not the CDN?**

**Reply by FeodorFitsner** [reply]
Alright, most probably GFW blocks/throttles the access to https://unpkg.com which hosts `canvaskit.wasm` required for CanvasKit renderer.

You have at least two options to try:

1) Switch app to use HTML renderer ([more info here](https://flet.dev/docs/controls/text/#using-system-fonts)):

```python
flet.app(target=main, view=flet.WEB_BROWSER, web_renderer="html")
```

2) Download `canvaskit.wasm` and put it in [app assets](https://flet.dev/docs/controls/image/#src) directory along with modified `index.html` ([more info here](https://stackoverflow.com/questions/70747972/what-is-the-reason-for-the-canvaskit-folder-in-flutter-wen-builds)):

```html
<script>
  window.flutterConfiguration = {
    canvasKitBaseUrl: "/canvaskit/"
  };
</script>
```

**Reply by jet10000** [reply]
https://www.jianshu.com/p/f799724b7a36

**Reply by FeodorFitsner** [reply]
Could you please explain what's that link about? :)

**Reply by jet10000** [reply]
Because flutter web app uses some network resources, such as fonts, but if you want to develop an application for LAN use only, you need to manually repair these connection resources, otherwise it will not work properly.

**Reply by jet10000** [reply]
see https://github.com/flutter/flutter/issues/60069

**Reply by FeodorFitsner** [reply]
So, it's fonts only the issue?

**Reply by jet10000** [reply]
Not just fonts

**Reply by jet10000** [reply]
https://user-images.githubusercontent.com/2258120/193730795-8192ed67-6370-4de7-9deb-2983aa405b69.mp4

**Reply by jet10000** [reply]
Remove google fonts and use the default fonts of the user phone&desktop OS system directly.

It's a suggestion.

**Reply by FeodorFitsner** [reply]
Would you suggest how?
